data/virt_autotest/virt_logs_collector.sh

```diff
@@ -0,0 +1,427 @@
+              echo -e "collect_extra_logs_via_guest_console ${guest_domain} ${logs_folder} ${extra_logs}"
+              collect_extra_logs_via_guest_console ${guest_domain} ${logs_folder} ${extra_logs}
+	      if [[ $? -eq 0 ]];then
+                 return 0
+              else
+                 return 1
+              fi
+           else
+	      return 0
+	   fi
+        else
+           echo -e "There is no extra logs to be collected. Please specify something."
+	   return 0
+        fi
+}
+
+#Collect any extra wanted logs from host. And provide more complete virtualization logs for SLE-11-SP4 and SLE-12 hosts.
+function collect_extra_logs_from_host() {
+	local logs_folder=$1
+	shift
+	local extra_logs=$@
+	local ret_result=0
+
+	if [[ ${extra_logs} != "" ]];then
+	   cp -f -r ${extra_logs} ${logs_folder}
+           ret_result=$?
+	   if [[ ${ret_result} -eq 0 ]];then
+	      echo -e "Successfully collected extra logs ${extra_logs} from host."
+	   else
+	      echo -e "Failed to collect extra logs ${extra_logs} from host."
+	   fi
+	else
+	   echo -e "No extra logs to be collected from host. Please specify something."
+	fi
+
+	local release=`get_sles_release release`
+	local spack=`get_sles_release spack`
+        local libvirt_boot_log="/var/lib/libvirt/boot"
+	local libvirt_qemu_log="/var/lib/libvirt/qemu"
+	local libvirt_log="/var/log/libvirt"
+	local libvirtd_log="${libvirt_log}/libvirtd.log"
+        local xen_log="/var/log/xen"
+	local xen_boot_log="${xen_log}/xen-boot.log"
+
+	if [[ ${release} -lt 12 ]];then
+           cp -f -r ${libvirt_log} ${logs_folder}
+	   ret_result=$(( ${ret_result} | $? ))
+           if [[ `get_sles_hypervisor` == "XEN" ]];then
+	      cp -f -r ${xen_log} ${logs_folder}
+	      ret_result=$(( ${ret_result} | $? ))
+           fi
+        elif [[ ${release} -eq 12 ]];then
+           cp -f -r ${libvirt_boot_log} ${logs_folder}
+           ret_result=$(( ${ret_result} | $? ))
+           cp -f -r ${libvirt_qemu_log} ${logs_folder}
+           ret_result=$(( ${ret_result} | $? ))
+           cp -f -r ${libvirtd_log} ${logs_folder}
+           ret_result=$(( ${ret_result} | $? ))
+           if [[ `get_sles_hypervisor` == "XEN" ]];then
+	      cp -f -r ${xen_boot_log} ${logs_folder}
+	      ret_result=$(( ${ret_result} | $? ))
+           fi           
+        fi
+
+	return ${ret_result}
+}
+
+#Usage and help info for the script
+help_usage(){
+	echo "script usage: $(basename $0) [-f \"Folder to be used as logs residence(Can be omitted/Default to /tmp/virt_logs_residence)\"] [-l \"Extra folders or files to be collected as host logs,for example,\"log_file_1 log_file_2 log_folder_1\"(Can be omitted/Default to nothing)\"] [-g \"guests to be involved or none,for example,\"guest1 guest2 guest3\"(Can be omitted/Default to all)\"] [-e \"Extra folders or files to be collected as guest logs, for example, \"log_file_1 log_file_2 log_folder_1\"(Can be omitted/Default to nothing)\"] [-h help]"
```

Comment:
This line is tooo long.

Reply:
This long help explain script usage with more details.

Reply:
I changed it to multi-line format.